crud/common/utils.lua

```diff
@@ -251,11 +253,46 @@ function utils.is_uuid(value)
     return ffi.istype(uuid_t, value)
 end
 
-function utils.format_result(rows, space)
-    return {
-        metadata = table.copy(space:format()),
-        rows = rows,
-    }
+local function get_field_metadata(full_metadata, field)
+    dev_checks('table', 'string')
+
+    local metadata = metadata_cache[full_metadata]
+    if metadata ~= nil then
+        return metadata[field]
+    end
+
+    metadata_cache[full_metadata] = {}
+    for _, tuple in ipairs(full_metadata) do
+        metadata_cache[full_metadata][tuple.name] = tuple
+    end
+
+    return metadata_cache[full_metadata][field]
+end
+
+local function format_metadata(full_metadata, fields)
+    dev_checks('table', 'table')
+
+    local metadata = {}
+
+    for i, field in ipairs(fields) do
+        metadata[i] = get_field_metadata(full_metadata, field)
+    end
+
+    return metadata
+end
+
+function utils.format_result(rows, space, fields)
+    local result = {}
+    local metadata = table.copy(space:format())
+    result.rows = rows
+
+    if fields ~= nil then
+        result.metadata = format_metadata(metadata, fields)
+    else
+        result.metadata = metadata
+    end
```

Comment:
I suggest renaming `metadata` to `space_format`. In fact, in becomes metadata only when we return it. But inside the function it's much more simple to think about `space format` than think about some `metadata`.

```lua
local space_format = table.copy(space:format())
result_fields_format = filter_format_fields(space_format)

result.metadata = result_fields_format
```

If this case function `filter_format_fields`(`format_metadata `) can check if `fields` option is specified and return full format if isn't.